src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1603,14 +1783,75 @@ private static void VerifyDerivedTypeConstraints(IEdmModel model, IEdmVocabulary
             throw new ODataException(Strings.PathParser_TypeCastOnlyAllowedInDerivedTypeConstraint(fullTypeName, kind, name));
         }
 
-        private static string ResolveEscapeFunction(string identifier, IEdmType bindingType, IEdmModel model, out string parenthesisExpression)
+        private bool TryResolveEscapeFunction(IEdmType bindingType, IEdmModel model, out string qualifiedName, out string parenthesisExpression, out bool anotherEscapeFunctionStarts)
         {

```

Comment:
Note that this function was originally written to be called within TryCreateSegmentForOperation, so its inputs/outputs may be a little different now that you are calling this before trying to bind the operation (you now have the actual function you need to bind).  So think about what input/output parameters make sense in this new usage, and whether it should be called from TryBindEscapeFunction or the logic should be merged with TryBindEscapeFunction.